code-challenges.test.js

```diff
@@ -13,40 +13,128 @@
+const remainArray = (array) => { // created a function called RemainArray that passes array as its parameter
+  const onlyNums = array.filter((value) => typeof value === 'number') // created a variable for the numbers only by using the filter method to create an array of only the values that are the datatype of 'number' 
+  const remainders = onlyNums.map((value) => value % 3) // take in the array of only the datatypes of 'number's and use the medulo operator to divide and get the remainders of the numbers inside the array
+    return remainders // returns the array only the remainders of the numbers
+}
+// console.log(remainArray(hodgepodge1))
+// console.log(remainArray(hodgepodge2))
 
-// a) Create a test with an expect statement using the variables provided.
+// output:  PASS  ./code-challenges.test.js
+// [ 2, 0, -1, 0 ]
+// [ 2, 1, -1 ]
 
-const cubeAndSum1 = [2, 3, 4]
-// Expected output: 99
-const cubeAndSum2 = [0, 5, 10]
-// Expected output: 1125
 
-// b) Create the function that makes the test pass.
+// --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// a) Create a test with an expect statement using the variables provided.
+describe("cubedArray", () => {
+  it("returns the sum of all the numbers cubed.", () => {
+      const cubeAndSum1 = [2, 3, 4]
+//     // Expected output: 99
+      const cubeAndSum2 = [0, 5, 10]
+//     // Expected output: 1125
+    expect(cubedArray(cubeAndSum1)).toEqual(99)
+    expect(cubedArray(cubeAndSum2)).toEqual(1125)
+  })
+})
+//Good Failure: ReferenceError: CubedArray is not defined
+
+// // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// Function name: CubedArray
+// Input: an array of numbers
+// Output: a number, the number is the total value of all the cubed sums
+// Process: create a function, create an empty array to hold the cubed number, create a variable for the sum to be held, then use a for loop to iterate through the array, create a variable to cube each index by itself, the use the .push method to move the cubed numbers into the empty array, then use the sum variable and reassign it to add up each of the cubed numbers, lastly have it return the summed up number of all the indexes cubed.
+
+
+const cubedArray = (array) => { //Created a new function that passes the parameter of array
+    let newArray = [] //created a new variable to hold an empty array
+    let sum = 0 //created a new variable of sum to hold the number (0 is placed to hold the data type number)
+    for (let i = 0; i < array.length; i++) { //for loop to iterate through the array
+      let cubed = array[i] * array[i] * array[i] //a variable called cubed stores the arrays indexes being cubed one after another
+      newArray.push(cubed) //takes the cubed numbers and pushes each number into the empty array
+      sum = newArray[0] + newArray[1] + newArray[2] // Adds up each number in the array, now that each number is cubed. 
```

Comment:
You want to keep your code dynamic. What happens if the array has more that 3 numbers? This function would not successfully get the sum at the end.